--- picojpeg-master/picojpeg.c.org	2017-11-12 16:45:33.257178839 +0900
+++ picojpeg-master/picojpeg.c	2017-11-12 16:47:20.559620404 +0900
@@ -1725,10 +1725,13 @@ static void convertCb(uint8 dstOfs)
       int16 cbG, cbB;
 
       cbG = ((cb * 88U) >> 8U) - 44U;
-      *pDstG++ = subAndClamp(pDstG[0], cbG);
+      pDstG[0] = subAndClamp(pDstG[0], cbG);
 
       cbB = (cb + ((cb * 198U) >> 8U)) - 227U;
-      *pDstB++ = addAndClamp(pDstB[0], cbB);
+      pDstB[0] = addAndClamp(pDstB[0], cbB);
+
+      ++pDstG;
+      ++pDstB;
    }
 }
 /*----------------------------------------------------------------------------*/
@@ -1746,10 +1749,13 @@ static void convertCr(uint8 dstOfs)
       int16 crR, crG;
 
       crR = (cr + ((cr * 103U) >> 8U)) - 179;
-      *pDstR++ = addAndClamp(pDstR[0], crR);
+      pDstR[0] = addAndClamp(pDstR[0], crR);
 
       crG = ((cr * 183U) >> 8U) - 91;
-      *pDstG++ = subAndClamp(pDstG[0], crG);
+      pDstG[0] = subAndClamp(pDstG[0], crG);
+
+      ++pDstR;
+      ++pDstG;
    }
 }
 /*----------------------------------------------------------------------------*/
