--- source/ffconf.h.org	2018-12-24 12:32:39.000000000 +0900
+++ source/ffconf.h	2018-12-24 12:36:28.000000000 +0900
@@ -1,3 +1,13 @@
+/*
+  Gadgetseed ffconf.h
+  for FatFs version 0.13b
+
+  2018.06.17 Takashi SHUDO
+*/
+
+#include "sysconfig.h"
+#include "task/mutex.h"
+
 /*---------------------------------------------------------------------------/
 /  FatFs Functional Configurations
 /---------------------------------------------------------------------------*/
@@ -37,8 +47,11 @@
 /* This option switches filtered directory read functions, f_findfirst() and
 /  f_findnext(). (0:Disable, 1:Enable 2:Enable with matching altname[] too) */
 
-
+#ifdef GSC_FATFS_ENABLE_MKFS
+#define FF_USE_MKFS		1
+#else
 #define FF_USE_MKFS		0
+#endif
 /* This option switches f_mkfs() function. (0:Disable or 1:Enable) */
 
 
@@ -50,7 +63,11 @@
 /* This option switches f_expand function. (0:Disable or 1:Enable) */
 
 
+#ifdef GSC_FATFS_ENABLE_CHMOD
+#define FF_USE_CHMOD	1
+#else
 #define FF_USE_CHMOD	0
+#endif
 /* This option switches attribute manipulation functions, f_chmod() and f_utime().
 /  (0:Disable or 1:Enable) Also FF_FS_READONLY needs to be 0 to enable this option. */
 
@@ -97,7 +114,7 @@
 */
 
 
-#define FF_USE_LFN		0
+#define FF_USE_LFN		2
 #define FF_MAX_LFN		255
 /* The FF_USE_LFN switches the support for LFN (long file name).
 /
@@ -137,7 +154,7 @@
 /  on character encoding. When LFN is not enabled, these options have no effect. */
 
 
-#define FF_STRF_ENCODE	3
+#define FF_STRF_ENCODE	0
 /* When FF_LFN_UNICODE >= 1 with LFN enabled, string I/O functions, f_gets(),
 /  f_putc(), f_puts and f_printf() convert the character encoding in it.
 /  This option selects assumption of character encoding ON THE FILE to be
@@ -163,7 +180,11 @@
 / Drive/Volume Configurations
 /---------------------------------------------------------------------------*/
 
+#ifdef GSC_FATFS_VOLUME_NUM
+#define FF_VOLUMES		GSC_FATFS_VOLUME_NUM
+#else
 #define FF_VOLUMES		1
+#endif
 /* Number of volumes (logical drives) to be used. (1-10) */
 
 
@@ -263,9 +284,9 @@
 
 
 /* #include <somertos.h>	// O/S definitions */
-#define FF_FS_REENTRANT	0
+#define FF_FS_REENTRANT	1
 #define FF_FS_TIMEOUT	1000
-#define FF_SYNC_t		HANDLE
+#define FF_SYNC_t		struct st_mutex *
 /* The option FF_FS_REENTRANT switches the re-entrancy (thread safe) of the FatFs
 /  module itself. Note that regardless of this option, file access to different
 /  volume is always re-entrant and volume control functions, f_mount(), f_mkfs()
