--- lwip-2.1.0/src/api/netdb.c.org	2017-12-17 15:08:10.425711221 +0900
+++ lwip-2.1.0/src/api/netdb.c	2017-12-17 15:09:27.972840817 +0900
@@ -47,7 +47,8 @@
 #include "lwip/dns.h"
 
 #include <string.h> /* memset */
-#include <stdlib.h> /* atoi */
+// for GadgetSeed #include <stdlib.h> /* atoi */
+int atoi(const char *nptr); // for GadgetSeed
 
 /** helper struct for gethostbyname_r to access the char* buffer */
 struct gethostbyname_r_helper {
